#!/usr/bin/env python
PACKAGE = "velocity_smoother"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("k_longitudinal", double_t, 0, "Low pass filter parameter K in longitudinal axis", .5, 0, 1)
gen.add("k_lateral", double_t, 0, "Low pass filter parameter K in lateral axis", .5, 0, 1)
gen.add("k_vertical", double_t, 0, "Low pass filter parameter K in vertical axis", .5, 0, 1)

gen.add("k_roll", double_t, 0, "Low pass filter parameter K in roll", .5, 0, 1)
gen.add("k_pitch", double_t, 0, "Low pass filter parameter K in pitch", .5, 0, 1)
gen.add("k_yaw", double_t, 0, "Low pass filter parameter K in yaw", .5, 0, 1)

gen.add("use_twist_stamped", bool_t, 0, "subscribe geometry_msgs/Twist if the value is False, subscribe geometry_msgs/TwistStamped if the value is True", True)

exit(gen.generate(PACKAGE, "velocity_smoother", "VelocitySmoother"))